### ISSUE TITLE
No caption for table?

### ISSUE THREAD

**Issue by miekg** [rfc-format/draft-iab-xml2rfc-v3-bis#31]
See https://trac.tools.ietf.org/tools/xml2rfc/trac/ticket/356 (closed because not a CLI issue)

There is no way to caption a table, texttable at least had postamble, but a table can't be contained in a figure (because it can contain a figure?).
This means I can't render a table with 'Table N' in the document, because you can't specify that in the XML.

Artwork and sourcecode don't suffer from this limitation.

**Comment by levkowetz** [comment]
Hi Miek,

On 2018-08-06 23:28, Miek Gieben wrote:
> See https://trac.tools.ietf.org/tools/xml2rfc/trac/ticket/356 (closed
> because not a CLI issue)
>
> There is no way to caption a table, texttable at least had postamble,
> but a table can't be contained in a figure (because it can contain a
> figure?). This means I can't render a table with 'Table N' in the
> document, because you can't specify that in the XML.

Something's not right here.  In v3 (RFC7991) all tables will get a "Table N"
caption, and if you provide a <name>, it will additionally contain the given
text: https://tools.ietf.org/html/rfc7991#section-2.54

I think what you want to do is perfectly possible with the v3 vocabulary.

Please provide the xml and the xml2rfc invocation you used to process it
so I can work out what's happening, and whether there's a bug in the
formatter.

> Artwork and sourcecode don't suffer from this limitation.

Not sure exactly what you mean here, but let's sort out the issue above
first.


	Henrik

**Comment by miekg** [comment]
But `<name>` is optional, and I don't have a (markdown) way to name tables anyway. I do have a way to add a table caption. I.e. I *just* want the caption and not the overhead of adding a name.

At least for artwork and source code I can wrap them in a figure: https://tools.ietf.org/html/rfc7991#section-2.25

and use `<postamble>`, but that is not allowed for the table element.

**Comment by levkowetz** [comment]
On 2018-08-07 00:10, Miek Gieben wrote:
> But `<name>` is optional, and I don't have a (markdown) way to name tables anyway. I do have a way to add a table caption. I.e. I *just* want the caption and not the overhead of adding a name.
>
> At least for artwork and source code I can wrap them in a figure: https://tools.ietf.org/html/rfc7991#section-2.25

The <name> _is_ just a caption.  It doesn't name anything in a deeper sense.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [rfc-format/draft-iab-xml2rfc-v..." ]
>
>
>On 2018-08-07 00:10, Miek Gieben wrote:
>> But `<name>` is optional, and I don't have a (markdown) way to name tables anyway. I do have a way to add a table caption. I.e. I *just* want the caption and not the overhead of adding a name.
>>
>> At least for artwork and source code I can wrap them in a figure: https://tools.ietf.org/html/rfc7991#section-2.25
>
>The <name> _is_ just a caption.  It doesn't name anything in a deeper sense.

ok.

So here is the inconsistency:

table -> <name> used for caption
artwork -> does not have <name>, needs to be wrapped in <figure> for <name> to exist
sourcecode -> same as artwork

It would also help if 7991 does not explicitly call out that a caption or forbidden:

> Deprecate <preamble> and <postamble> in favor of simply using <t>
> before or after the figure.  This also deprecates the "align"
> attribute in <figure>.

This not true, because `<name>` is the new captioning mechanism?

**Comment by reschke** [comment]
On 2018-08-07 00:02, Henrik Levkowetz wrote:
> ... > Something's not right here. In v3 (RFC7991) all tables will get a 
"Table N"
> caption, and if you provide a <name>, it will additionally contain the given
> text: https://tools.ietf.org/html/rfc7991#section-2.54
> ...

FWWIW: <https://github.com/rfc-format/draft-iab-xml2rfc-v3-bis/issues/14>

**Comment by levkowetz** [comment]
> On 7 Aug 2018, at 05:28, Julian Reschke <notifications@github.com> wrote:
> 
> On 2018-08-07 00:02, Henrik Levkowetz wrote:
> > ... > Something's not right here. In v3 (RFC7991) all tables will get a 
> "Table N"
> > caption, and if you provide a <name>, it will additionally contain the given
> > text: https://tools.ietf.org/html/rfc7991#section-2.54
> > ...
> 
> FWWIW: <https://github.com/rfc-format/draft-iab-xml2rfc-v3-bis/issues/14>

I would think bare <artwork> (not within a <figure>) would provide what issue 14 asks for?

Best,

    Henrik

**Comment by reschke** [comment]
Possibly, but that leaves the question open for tables...

**Comment by miekg** [comment]
Ok, so
```
<table>
<thead>
<tr>
<th>Name</th>
<th>Age</th>
</tr>
</thead>

<tbody>
<tr>
<td>Bob</td>
<td>27</td>
</tr>

<tr>
<td>Alice</td>
<td>23</td>
</tr>
</tbody>
</table>
```
Generates:
```

   +-------+-----+
   | Name  | Age |
   +=======+=====+
   | Bob   | 27  |
   +-------+-----+
   | Alice | 23  |
   +-------+-----+

                                  Table 1
```
From the spec I can't tell if this is correct or not.

**Comment by levkowetz** [comment]
I agree that the current placement of the caption doesn't look good with a narrow table.

However, I'd argue that it would be best to centre the caption under the table, and then align the whole (table with caption) according to a new align keyword on table.

**Comment by levkowetz** [comment]
On 2018-08-07 00:42, Miek Gieben wrote:
>
> This not true, because `<name>` is the new captioning mechanism?

There's a difference, in that <preamble> and <postamble> added paragraphs,
rather than captions.  In v2, the 'title' attribute on <figure> provided
the caption; in v3 it is provided by <name>.  Same for <texttable>/<table>.

**Comment by paulehoffman** [comment]
Consensus on the list is to implement this, but I'm unclear how. Message asking about this sent to the list.

**Comment by paulehoffman** [comment]
Closed with PR #68
